src/web/lib/getCountryCode.ts

```diff
@@ -1,76 +1,56 @@
 
-function hasExpired(whenItExpires: number) {
-	return new Date().getTime() > whenItExpires;
-}
-
-export const setCountryCode = (countryCode: string): void => {
-	// What's this setTimeout business?
-	// localStorage calls are syncronous and we don't need to wait for this
-	// one so we use setTimeout to put this step on the end of the event queue
-	// for later, letting the thread continue
-	setTimeout(() => {
-		if (countryCode) {
-			try {
-				localStorage.setItem(
-					COUNTRY_CODE_KEY,
-					JSON.stringify({
-						value: countryCode,
-						expires: new Date().getTime() + TEN_DAYS,
-					}),
-				);
-			} catch (error) {
-				// We tried, it failed. Often local storage is not available and we
-				// need to live with that
-			}
-		}
-	});
-};
+const COUNTRY_CODE_KEY = 'GU_geo_country';
+const COUNTRY_CODE_KEY_OVERRIDE = 'gu.geo.override';
+/*
+	Memoizes the location
+ */
+let locale: CountryCode | null;
 
-export const getCountryCode = async (): Promise<string | void> => {
-	// Read local storage to see if we already have a value
-	let localCountryCode: LocalCountryCodeType | null;
-	try {
-		const item = localStorage.getItem(COUNTRY_CODE_KEY);
-		localCountryCode = item ? JSON.parse(item) : null;
-	} catch (error) {
-		localCountryCode = null;
+/*
+	Can be used to override country code and sets it in localStorage
+ */
+export const overrideCountryCode = (countryCode: CountryCode): void => {
+	if (countryCode) {
+		storage.local.set(COUNTRY_CODE_KEY_OVERRIDE, countryCode);
+		locale = countryCode;
 	}
+};
 
-	if (!localCountryCode || hasExpired(localCountryCode.expires)) {
-		const countryCode = await fetch(
-			'https://api.nextgen.guardianapps.co.uk/geolocation',
-		)
-			.then((response) => {
-				if (!response.ok) {
-					throw Error(
-						response.statusText ||
-							`getCountryCode | An api call returned HTTP status ${response.status}`,
-					);
-				}
-				return response;
-			})
-			.then((response) => response.json())
-			.then((json: { country: string }) => json.country)
-			.catch((error) => {
-				window.guardian.modules.sentry.reportError(
-					error,
-					'get-country-code',
-				);
-			});
+/*
+	This method can be used as a non async way of getting the country code
+	after getLocaleCode has been called.
+ */
+export const getCountryCode = (): string | null => {
+	return (
+		locale ??
+		storage.local.get(COUNTRY_CODE_KEY_OVERRIDE) ??
+		getCookie(COUNTRY_CODE_KEY)
+	);
+};
 
-		if (countryCode) {
-			setCountryCode(countryCode);
-		}
-		// Return the country value that we got from our fetch call
-		return countryCode;
-	}
-	// THere was no need to fetch, return the local value
-	return localCountryCode.value;
+/*
+	This method returns the location of the user from guardian/libs getLocale
+	It will return fastly's 'GU_geo_country' if it exists,
+	or an overridden geolocation from 'gu.geo.override' localStorage
+	or if none of those exists, it will call the geo endpoint to fetch it and set it in `GU_geo_country`
```

Comment:
It will only set the cookie to the response of fetching from geolocation endpoint

> if none of those exists, it will call the geo endpoint to fetch it and set it in `GU_geo_country`

does it not make sense?

Reply:
https://github.com/guardian/libs/blob/698069356d5cdf7c1bf2caa62079aac6b183f08c/src/getLocale.ts#L49

Reply:
> Would you want to return the memoized locale if non-null?
Do you mean inside the catch callback? In this case the locale will be null and we have getting the cookie as fallback

Reply:
Sorry, I think the memoized `locale` is only for the synchronous version. Makes sense.